File: js/programmers/level.0/prgm_120924_1.js
function solution(common) {
  let gap = [];
  let result = 0;
  for (i = 0; i < 2; i++) {
    gap.push(common[i + 1] - common[i]);
  }
  if (gap[0] === gap[1]) {
    return (result = common[common.length - 1] + gap[0]);
  } else {
    return (result = common[common.length - 1] * (gap[1] / gap[0]));
  }
}

File: python/programmers/level.0/prgm_120924_1.py
def solution(common):
    gap = []

    for i in range(2):
        gap.append(common[i+1] - common[i])

    if gap[0] == gap[1]:
        return common[-1] + gap[0]
    else:
        return common[-1] * (gap[1] / gap[0])